lib/matplotlib/scale.py

```diff
@@ -220,11 +229,17 @@ def __init__(self, axis, **kwargs):
         if axis.axis_name == 'x':
             base = kwargs.pop('basex', 10.0)
             subs = kwargs.pop('subsx', None)
-            nonpos = kwargs.pop('nonposx', 'mask')
+            nonpos = kwargs.pop('nonposx', 'clip')
         else:
             base = kwargs.pop('basey', 10.0)
             subs = kwargs.pop('subsy', None)
-            nonpos = kwargs.pop('nonposy', 'mask')
+            nonpos = kwargs.pop('nonposy', 'clip')
+
+        if len(kwargs):
+            raise ValueError(("provided too many kwargs, can only pass "
+                              "{'basex', 'subsx', nonposx'} or "
+                              "{'basey', 'subsy', nonposy'}.  You passed ") +
+                             "{!r}".format(kwargs))
```

Comment:
just format the entire string?  no need to add, also no need for `!r` as `repr(dict) == format(dict, "")`

Reply:
If you try to format the whole string it blows up on a key error with "'basex', 'subsx', nonposx'".

Reply:
"{" -> "{{".
If you think it's not an improvement, then at least replace "{!r}.format(...)" by repr(...), no?  seems a bit perverse otherwise :)